tests/test_hash_is_constant.py

```diff
@@ -41,27 +41,47 @@ def setUp(self):
 	def test_average_hash(self):
 		result_hash = imagehash.average_hash(self.image)
 		known_hash = 'ffd7918181c9ffff'
+		known_bytes = b'\xff\xd7\x91\x81\x81\xc9\xff\xff'
+		known_bigint = -11380488630829057
 		self.assertEqual(str(result_hash), known_hash)
+		self.assertEqual(bytes(result_hash), known_bytes)
+		self.assertEqual(int(result_hash), known_bigint)
 
 	def test_phash(self):
 		result_hash = imagehash.phash(self.image)
 		known_hash = 'ba8c84536bd3c366'
+		known_bytes = b'\xba\x8c\x84\x53\x6b\xd3\xc3\x66'
+		known_bigint = -5004479592089271450
 		self.assertEqual(str(result_hash), known_hash)
+		self.assertEqual(bytes(result_hash), known_bytes)
+		self.assertEqual(int(result_hash), known_bigint)
 
 	def test_dhash(self):
 		result_hash = imagehash.dhash(self.image)
 		known_hash = '0026273b2b19550e'
+		known_bytes = b'\x00\x26\x27\x3b\x2b\x19\x55\x0e'
+		known_bigint = 10739184194639118
 		self.assertEqual(str(result_hash), known_hash)
+		self.assertEqual(bytes(result_hash), known_bytes)
+		self.assertEqual(int(result_hash), known_bigint)
 
 	def test_whash(self):
 		result_hash = imagehash.whash(self.image)
 		known_hash = 'ffd391818181a5e7'
+		known_bytes = b'\xff\xd3\x91\x81\x81\x81\xa5\xe7'
+		known_bigint = -12506388542413337
 		self.assertEqual(str(result_hash), known_hash)
+		self.assertEqual(bytes(result_hash), known_bytes)
+		self.assertEqual(int(result_hash), known_bigint)
 
 	def test_color_hash(self):
 		result_hash = imagehash.colorhash(self.image)
 		known_hash = '07007000000'
+		known_bytes = b'\x1c\x01\xc0\x00\x00\x00'
```

Comment:
The difference of the str hash and the bytes representation here is due to [numpy.packbits](https://numpy.org/doc/stable/reference/generated/numpy.packbits.html) padding to full bytes by inserting zero bits at the end.

The binary value is '00.01110000.00000111.00000000.00000000.00000000'. The _binary_array_to_hex function converts that to '0.70.07.00.00.00' (as if adding extra 0s to the beginning). But numpy.packbits inserts 0s at the end until it becomes full bytes, so it becomes '00011100.00000001.11000000.00000000.00000000.00000000' and is converted to '1c.01.c0.00.00.00'.

Reply:
To accommodate this, the bytes dunder method implementation could be updated to:

```python
if self.hash.size % 8 == 0:
    return numpy.packbits(self.hash).tobytes()
return numpy.packbits(numpy.pad(self.hash.flatten(), pad_width=(8 - (self.hash.size % 8), 0), mode='constant')).tobytes()
```

but I'd argue this is over-engineering just to match one arbitrary representation to another. Even so, bytes and _binary_array_to_hex won't produce the same output, as the former outputs 2-chars for every byte and the latter 1-char for every 4-bits.